### PR TITLE
feat(add): strip package subpath when adding a package

### PR DESCRIPTION
**Pull request by bartlomieju**
These now works:
```
$ deno add @std/dotenv/load
$ deno add npm:preact/hooks
```
Previously we were erroring out, because this is a "package reference" including
a subpath.

Closes https://github.com/denoland/deno/issues/25385

### CODE REVIEW

**Review by marvinhagemeister — APPROVED** [review]
Great idea!

File: cli/tools/registry/pm.rs [hunk 1]
```diff
@@ -501,14 +503,18 @@ impl AddPackageReq {
...
         Ok(AddPackageReq {
           alias: maybe_alias.unwrap_or_else(|| package_req.name.to_string()),
           value: AddPackageReqValue::Jsr(package_req),
         })
       }
       Prefix::Npm => {
-        let package_req = PackageReq::from_str(entry_text)?;
+        let npm_package_req =
+          NpmPackageReqReference::from_str(&format!("npm:{}", entry_text))?;
+        let package_req = npm_package_req.req().clone();
```
Comment (dsherret) [inline cli/tools/registry/pm.rs:517]:
```suggestion
        let req_ref =
          NpmPackageReqReference::from_str(&format!("npm:{}", entry_text))?;
        let package_req = req_ref.into_inner().req;
```

File: cli/tools/registry/pm.rs [hunk 2]
```diff
@@ -501,14 +503,18 @@ impl AddPackageReq {
 
     match prefix {
       Prefix::Jsr => {
-        let package_req = PackageReq::from_str(entry_text)?;
+        let jsr_package_req =
+          JsrPackageReqReference::from_str(&format!("jsr:{}", entry_text))?;
+        let package_req = jsr_package_req.req().clone();
```
Comment (dsherret) [inline cli/tools/registry/pm.rs:508]:
```suggestion
        let req_ref =
          JsrPackageReqReference::from_str(&format!("jsr:{}", entry_text))?;
        let package_req = req_ref.into_inner().req;
```

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**